public/js/grid/timetable_util.js

```diff
@@ -171,15 +175,16 @@ function saveCookies(courses, sections) {
     'use strict';
 
     if (courses !== undefined) {
-        setCookie("selected-courses", JSON.stringify(courses));
+        setCookie("selected-courses", JSON.stringify(courses).replace(/,/g, '_').replace(/[\"\[\]]/g, ''));
```

Comment:
Similarly, it should be possible to just do `courses.join('_')` rather than use JSON.